template/build/dev-server.js

```diff
@@ -1,3 +1,5 @@
+if (!process.env.NODE_ENV) process.env.NODE_ENV = 'development'
```

Comment:
I'm thinking since it's the dev server, `NODE_ENV` should always be overwritten here. Could we also place it a little lower and use [the value from the config](https://github.com/vuejs-templates/webpack/blob/master/template/config/index.js#L20)? For example:

``` js
process.env.NODE_ENV = config.dev.env
```